conf/configuration.go

```diff
@@ -83,67 +80,54 @@ type Configuration struct {
 	} `json:"external"`
 }
 
-// LoadGlobalFromFile loads global configuration from the provided filename.
-func LoadGlobalFromFile(name string) (*GlobalConfiguration, error) {
-	cmd := &cobra.Command{}
-	config := ""
-	cmd.Flags().StringVar(&config, "config", "config.test.json", "Config file")
-	cmd.Flags().Set("config", name)
-	return LoadGlobal(cmd)
+func loadEnvironment(filename string) error {
+	// default to .env if an empty string is passed as a filename
+	// godotenv will only default if passed an empty argument list
+	filenames := []string{}
+	if filename != "" {
+		filenames = append(filenames, filename)
+	}
+	if err := godotenv.Load(filenames...); err != nil {
+		// handle if .env file does not exist, this is OK
+		if os.IsNotExist(err) && filename == "" {
+			return nil
+		}
+		return err
+	}
+	return nil
 }
 
 // LoadGlobal loads configuration from file and environment variables.
-func LoadGlobal(cmd *cobra.Command) (*GlobalConfiguration, error) {
-	config := new(GlobalConfiguration)
-
-	if err := nconf.LoadConfig(cmd, "gotrue", config); err != nil {
+func LoadGlobal(filename string) (*GlobalConfiguration, error) {
+	if err := loadEnvironment(filename); err != nil {
 		return nil, err
 	}
 
-	if config.DB.ConnURL == "" && os.Getenv("DATABASE_URL") != "" {
-		config.DB.ConnURL = os.Getenv("DATABASE_URL")
-	}
-
-	if config.API.Port == 0 && os.Getenv("PORT") != "" {
-		port, err := strconv.Atoi(os.Getenv("PORT"))
-		if err != nil {
-			return nil, errors.Wrap(err, "formatting PORT into int")
-		}
-
-		config.API.Port = port
-	}
-
-	if config.API.Port == 0 && config.API.Host == "" {
-		config.API.Port = 8081
+	config := new(GlobalConfiguration)
+	if err := envconfig.Process("gotrue", config); err != nil {
+		return nil, err
 	}
-
 	if _, err := nconf.ConfigureLogging(&config.Logging); err != nil {
 		return nil, err
 	}
-
 	return config, nil
 }
 
-// LoadConfigFromFile loads per-instance configuration from the provided filename.
-func LoadConfigFromFile(name string) (*Configuration, error) {
-	cmd := &cobra.Command{}
-	config := ""
-	cmd.Flags().StringVar(&config, "config", "config.test.json", "Config file")
-	cmd.Flags().Set("config", name)
-	return LoadConfig(cmd)
-}
-
 // LoadConfig loads per-instance configuration.
-func LoadConfig(cmd *cobra.Command) (*Configuration, error) {
-	config := new(Configuration)
-	if err := nconf.LoadConfig(cmd, "gotrue", config); err != nil {
+func LoadConfig(filename string) (*Configuration, error) {
+	if err := loadEnvironment(filename); err != nil {
 		return nil, err
 	}
 
+	config := new(Configuration)
+	if err := envconfig.Process("gotrue", config); err != nil {
+		return nil, err
+	}
 	config.ApplyDefaults()
 	return config, nil
 }
 
+// ApplyDefaults sets defaults for a Configuration
```

Comment:
We need to be able to apply the defaults when deserializing from the database too.